bot/exts/info/code_snippets.py

```diff
@@ -18,22 +18,21 @@
 )
 
 GITHUB_GIST_RE = re.compile(
-    r'https://gist\.github\.com/([^/]+)/(?P<gist_id>[^\W_]+)/*'
-    r'(?P<revision>[^\W_]*)/*#file-(?P<file_path>\S+?)'
+    r'https://gist\.github\.com/([^/]+)/(?P<gist_id>[a-zA-Z0-9]+)/*'
+    r'(?P<revision>[a-zA-Z0-9-]*)/*#file-(?P<file_path>[^#>]+?)'
     r'(-L(?P<start_line>\d+)([-~:]L(?P<end_line>\d+))?)'
 )
 
 GITHUB_HEADERS = {'Accept': 'application/vnd.github.v3.raw'}
 
 GITLAB_RE = re.compile(
-    r'https://gitlab\.com/(?P<repo>\S+?)/\-/blob/(?P<path>\S+/[^\s#,>]+)'
-    r'(#L(?P<start_line>\d+)([-](?P<end_line>\d+))?)'
+    r'https://gitlab\.com/(?P<repo>[a-zA-Z0-9-]+?)/\-/blob/(?P<path>[^#>]+/{0,1})'
```

Comment:
What's the reason for doing `/{0,1}`? None of the other URLs seem to need it. Does GitLab do something extra which I'm not yet seeing?

Reply:
It's so that it matches `<part of the URL>/` for all parts of the path except the last, in which case it matches `<part of the URL>` only. I also do this in the GitHub regex. Is this part unnecessary?

Reply:
Sorry for getting back to you a bit late. I still don't understand what you mean. I thought you meant that it avoids matching a trailing slash in the path, but the regex still matches it. Here is a real example:

```
https://gitlab.com/gitlab-org/gitlab/-/blob/master/scripts/create_postgres_user.sh/#L3-4
```

Furthermore, GitLab resolves that URL just fine in my browser (though maybe it's cause my browser is silently stripping it). If the trailing slash is really a problem when you need to fetch the content, then using `rstrip("/")` in Python is a simple solution.

If you were not referring to a trailing slash, then please clarify.

Reply:
Oops, I just misunderstood what my regex did. You're right - the `/{0,1}` is unnecessary because the trailing slash is already matched in the preceding regex. I tested it and the slash isn't a problem when I fetch the content